spec/integration/application/agent_spec.rb

```diff
@@ -326,4 +326,75 @@
       end
     end
   end
+
+  context 'multiple agents running' do
+    it "exits if an agent is already running" do
+      server.start_server do |port|
+        path = Puppet[:agent_catalog_run_lockfile]
+        Puppet[:masterport] = port
+
+        th = Thread.new {
+          %x{ruby -e "File.write('#{path}', Process.pid); sleep(2); 'puppet keyword needed here'"}
+        }
+
+        until File.exists?(path) && File.size(path) > 0 do
+          sleep 0.1
+        end
+
+        expect {
+          agent.command_line.args << '--test'
+          agent.run
+        }.to exit_with(1).and output(/Run of Puppet configuration client already in progress; skipping/).to_stdout
+
+        th.kill # kill thread so we don't wait too much
+      end
+    end
+
+    it "waits for other agent run to finish before starting" do
+      server.start_server do |port|
+        path = Puppet[:agent_catalog_run_lockfile]
+        Puppet[:masterport] = port
+        Puppet[:waitforlock] = 1
```

Comment:
`waitforlock` is a duration which unfortunately requires an `Integer` value, unlike ttl which accepts `Numeric` which includes `Float`... We could change `waitforlock` to be a `:ttl` or allow `:duration` to accept `Numeric` instead of `Integer`. But it removes 1 second from the test so I think it's ok as is.